pkg/collector/corechecks/servicediscovery/servicediscovery.go

```diff
@@ -128,7 +153,124 @@ func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instance
 
 // Run executes the check.
 func (c *Check) Run() error {
-	return c.os.DiscoverServices()
+	start := time.Now()
+	defer func() {
+		diff := time.Since(start).Seconds()
+		metricTimeToScan.Observe(diff)
+	}()
+
+	disc, err := c.os.DiscoverServices()
+	if err != nil {
+		telemetryFromError(err)
+		return err
+	}
+
+	log.Debugf("aliveProcs: %d | ignoreProcs: %d | runningServices: %d | potentials: %d | openPorts: %s",
+		disc.aliveProcsCount,
+		len(disc.ignoreProcs),
+		len(disc.runningServices),
+		len(disc.potentials),
+		disc.openPorts.String(),
+	)
+	metricDiscoveredServices.Set(float64(len(disc.runningServices)))
+
+	runningServicesByName := map[string][]*serviceInfo{}
+	for _, svc := range disc.runningServices {
+		runningServicesByName[svc.meta.Name] = append(runningServicesByName[svc.meta.Name], svc)
+	}
+	for _, svcs := range runningServicesByName {
+		if len(svcs) <= 1 {
+			continue
+		}
+		for _, svc := range svcs {
+			if c.sentRepeatedEventPIDs[svc.process.PID] {
+				continue
+			}
+			err := fmt.Errorf("found repeated service name: %s", svc.meta.Name)
```

Comment:
That situation can happen quite often

Reply:
What scenarios make it likely? We do want to know how many duplicate service names we generate so we can see if we need to tweak the service name generation code.